src/test/java/uk/gov/hmcts/reform/managecase/service/noc/PrepareNoCServiceTest.java

```diff
@@ -0,0 +1,366 @@
+            assertEquals("[Defendant]", caseRoleId.findPath("list_items").get(1).findPath("code").textValue());
+            assertEquals("Defendant", caseRoleId.findPath("list_items").get(1).findPath("label").textValue());
+        }
+
+        @Test
+        @DisplayName("PrepareNoC event successfully returned for a caseworker user")
+        void shouldReturnPrepareNoCEventForCaseworker() {
+            mockUserCaseRolesForCaseworker();
+
+            Map<String, JsonNode> caseData = new HashMap<>();
+            orgPolicies = prepareOrganisationPoliciesForCaseworkerTest();
+            orgPolicies.forEach((key, value) -> caseData.put(key, organisationPolicyToJsonNode(value)));
+            caseData.put(COR_FIELD_NAME, changeOrganisationRequestToJsonNode(changeOrganisationRequest()));
+
+            CaseDetails caseDetails = new CaseDetails(REFERENCE, JURISDICTION, "", CASE_TYPE, caseData,
+                                                      securityClassification, dataClassification);
+
+            Map<String, JsonNode> result = prepareNoCService.prepareNoCRequest(caseDetails);
+
+            JsonNode caseRoleId = result.get(COR_FIELD_NAME).findPath(COR_CASE_ROLE_ID);
+            assertEquals("[Claimant]", caseRoleId.findPath("value").findPath("code").textValue());
+            assertEquals("Claimant", caseRoleId.findPath("value").findPath("label").textValue());
+            assertTrue(caseRoleId.findPath("list_items").isArray());
+            assertEquals(2, caseRoleId.findPath("list_items").size());
+            assertEquals("[Claimant]", caseRoleId.findPath("list_items").get(0).findPath("code").textValue());
+            assertEquals("Claimant", caseRoleId.findPath("list_items").get(0).findPath("label").textValue());
+            assertEquals("[Defendant]", caseRoleId.findPath("list_items").get(1).findPath("code").textValue());
+            assertEquals("Defendant", caseRoleId.findPath("list_items").get(1).findPath("label").textValue());
+        }
+
+        @Test
+        @DisplayName("should error when there is no Organisation policy records on the case")
+        void shouldErrorWhenThereIsNoOrganisationPolicyRecordsOnTheCase() {
+            mockUserCaseRolesForSolicitor();
+
+            Map<String, JsonNode> caseData = new HashMap<>();
+            caseData.put(COR_FIELD_NAME, changeOrganisationRequestToJsonNode(changeOrganisationRequest()));
+            CaseDetails caseDetails = new CaseDetails(REFERENCE, JURISDICTION, "", CASE_TYPE, caseData,
+                                                      securityClassification, dataClassification);
+
+            ValidationException exception = assertThrows(
+                ValidationException.class,
+                () -> prepareNoCService.prepareNoCRequest(caseDetails)
+            );
+            assertThat(
+                exception.getMessage(),
+                is("No OrganisationPolicy found on the case data.")
```

Comment:
If string should match validation error then import and match on constant defined in *src/main/java/uk/gov/hmcts/reform/managecase/api/errorhandling/ValidationError.java*.